### ISSUE TITLE
tiny sin

### ISSUE THREAD

**Issue by alto777** [AScustomWorks/AS#21]
hi Alfredo!

I noticed your update and came to look. I am sure it was not your intention, but seeing your modules takes the wind out of my sails… nice stuff!

I was intrigued by your comment about correcting the sin wave output of the "MyModule" Template plugin. I looked at both your version and the original, and I do not see a difference.

You: sine = sinf(2.0f * M_PI * (phase+1 * 0.125f))

which is just a phase shift of the original, unless you meant

sine = sinf(2.0f * M_PI * ((phase+1) * 0.125f))

which is just a frequency change.

I don't know where sine = sinf(2 * M_PI * phase)+ sinf(2 * M_PI * phase * 2)*5 came from, unless it was in a version of MyModule.cpp that has come and gone. Certainly not a pure sin wave! But it's just a sin wave plus a heavily weighted harmonic, unless it was

sine = (sinf(2 * M_PI * phase)+ sinf(2 * M_PI * phase * 2))*5

in which case again we have a sin wave and harmonic which is a pleasant sound sawfish.

As for the polarity at a glance all the functions have a range of -5 to 5 after the scaling multiplication.

BTW I found https://www.desmos.com/calculator very useful for looking at this close.y.

alto777

**Comment by AScustomWorks** [comment]
Hi Robert, thanks!
Both osc modules were meant to be just tests to learn my way around vcv rack so I never got back to really review the code, let alone them having any accuracy hehe.
 Indeed the original  came from an old version of the template example, and someone else told me about the not so pure sine wave... so I found the actual working code somewhere else, but still, to me it was only ear and what I saw on the scope. I been thinking about switching to the minBLEP version though.

Thanks to you now I know a bit more about the subject, and the new (to me) info will come handy in the future for sure ;)
